Simulator/src/main/kotlin/edu/drexel/se575/BlockChain.kt

```diff
@@ -14,13 +17,14 @@ class BlockChain(var blockList: ArrayList<Block> = arrayListOf()) {
         }
     }
 
-    fun addTransactionToQueue(transaction: Transaction){
+    fun addTransactionToQueue(transaction: Transaction) {
+        interpreter.runContract(transaction.data)
         transactionQueue.addTransaction(transaction)
     }
 
 
     //TODO replace empty validator/signature values
```

Comment:
If I'm thinking properly, the validator should take the **public** key of the signer, and the signature is just the signature they applied. We have everything here for this now.

Reply:
@Romulus10 I wasn't sure if validator would be the public key or the human readable account name so we can get the associated public key. Passing public key was what I was leaning towards.